### ISSUE TITLE
Custom Trait-Attribute causes non-stop Discovery of Test-Methods in VS 2019

### ISSUE THREAD

**Issue by lanluu** [xunit/xunit#2025]
Hi,

in VS 2019 I am suffering a situation in that the discovery of xUnit Test-Methods in Test-Explorer (i.e. test_host process) runs endless, if any Test-Method has Custom-Trait-Attribute i.e. by adding any class inhered from ITraitAttribute and ITraitDiscoverer, for instance: https://github.com/xunit/samples.xunit/tree/master/TraitExtensibility

I would appreciate any idea in this case.

Thanks and best regards!

**Comment by lanluu** [comment]
As I figured out, this situation can caused in case the Custom-Trait-Attribute throws any Exception in Constructor e.g. because of validation of input params. For instance:

```
    [TraitDiscoverer("...", "...")]
    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
    public class ClassificationAttribute : Attribute, ITraitAttribute
    {
        public ClassificationAttribute(params Type[] traits)
        {
            // if you throw any Exception here because of invalid parameters,
            // and any Test-Method has this Attribute with invalid parameters,
            // then testhost process (in Test-Explorer) will run endless.
        }
    }
```

I hope it might help someones.

**Comment by bradwilson** [comment]
Does this only fail in Visual Studio? If so, this issue needs to be moved to the right repo.

**Comment by lanluu** [comment]
Yes, only in Visual Studio. To which Repo should I move this thread?

**Comment by bradwilson** [comment]
I'll move it. :)
